pkg/operation/operation.go

```diff
@@ -371,6 +373,54 @@ func (o *Operation) ChartInitializer(chartName string, values map[string]interfa
 	}
 }
 
+// SyncClusterResourceToSeed creates or updates the `Cluster` extension resource for the shoot in the seed cluster.
+// It contains the shoot, seed, and cloudprofile specification.
+func (o *Operation) SyncClusterResourceToSeed(ctx context.Context) error {
+	if err := o.InitializeSeedClients(); err != nil {
+		o.Logger.Errorf("Could not initialize a new Kubernetes client for the seed cluster: %s", err.Error())
+		return err
+	}
+
+	var (
+		cloudProfileObj = o.Shoot.CloudProfile.DeepCopy()
+		seedObj         = o.Seed.Info.DeepCopy()
+		shootObj        = o.Shoot.Info.DeepCopy()
+	)
+
+	cloudProfileObj.TypeMeta = metav1.TypeMeta{
+		APIVersion: gardenv1beta1.SchemeGroupVersion.String(),
+		Kind:       "CloudProfile",
+	}
+	seedObj.TypeMeta = metav1.TypeMeta{
+		APIVersion: gardenv1beta1.SchemeGroupVersion.String(),
+		Kind:       "Seed",
+	}
+	shootObj.TypeMeta = metav1.TypeMeta{
+		APIVersion: gardenv1beta1.SchemeGroupVersion.String(),
+		Kind:       "Shoot",
+	}
+
+	return o.ChartApplierSeed.ApplyChart(ctx, filepath.Join(common.ChartPath, "seed-cluster"), "", o.Shoot.SeedNamespace, nil, map[string]interface{}{
```

Comment:
I also think charts are rather an overkill here - If you just create a cluster resource in the target cluster, you should have the same effect + you shouldn't need to set the `APIVersion` + `Kind` since this is done already by the client.

Reply:
Agreed to create the resource directly, however, the `TypeMeta` is not added automatically.